From 04d8e3914f32cfe9b34fe6586ede26bb5cb4ff96 Mon Sep 17 00:00:00 2001
From: Michael Scott <mike@foundries.io>
Date: Mon, 8 May 2023 16:33:00 -0700
Subject: [PATCH 1/3] imx8mm_var_dart: runtime image offset detection for
 imx8mm

Add support of next image ram offset detection (in SPL) for
IMX8MM_VAR_DART.

Signed-off-by: Michael Scott <mike@foundries.io>
---
 include/configs/imx8mm_var_dart.h | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/include/configs/imx8mm_var_dart.h b/include/configs/imx8mm_var_dart.h
index 9368674a361..a69ad7a2aa3 100644
--- a/include/configs/imx8mm_var_dart.h
+++ b/include/configs/imx8mm_var_dart.h
@@ -15,6 +15,16 @@
 #define CONFIG_SYS_BOOTM_LEN		(64 * SZ_1M)
 #define CONFIG_SPL_MAX_SIZE		(148 * 1024)
 #define CONFIG_SYS_MONITOR_LEN		SZ_512K
+/*
+ * CONFIG_SECONDARY_BOOT_SECTOR_OFFSET was intended to be used as a compile time setting.
+ * If CONFIG_SECONDARY_BOOT_RUNTIME_DETECTION is defined, then only use the base offset of 0x300 here.
+ * CONFIG_SECONDARY_BOOT_SECTOR_OFFSET will be added dynamically if needed.
+ */
+#ifndef CONFIG_SECONDARY_BOOT_RUNTIME_DETECTION
+#undef CONFIG_SYS_MMCSD_RAW_MODE_U_BOOT_SECTOR
+#define CONFIG_SYS_MMCSD_RAW_MODE_U_BOOT_SECTOR (0x300 + CONFIG_SECONDARY_BOOT_SECTOR_OFFSET)
+#endif /* CONFIG_SECONDARY_BOOT_RUNTIME_DETECTION */
+
 #define CONFIG_SYS_UBOOT_BASE	\
 	(QSPI0_AMBA_BASE + CONFIG_SYS_MMCSD_RAW_MODE_U_BOOT_SECTOR * 512)
 
-- 
2.40.1

